packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -54,7 +54,9 @@ export async function createNewFluidFile(
     }
 
     const filePath = newFileInfo.filePath ? encodeURIComponent(`/${newFileInfo.filePath}`) : "";
-    const encodedFilename = encodeURIComponent(`${newFileInfo.filename}.fluid`);
+    const fileName = newFileInfo.fileExtension
```

Comment:
So is it required or not required argument?
If it's required argument, I'd remove this check.
If not, can you please test for undefined, not for ""?

Reply:
Actually, can we undo ALL changes, and just remove .fluid ?
I.e. we say driver has no default. Whatever filename brings is what we use. If it has no extension, then file will be with no extension. I think that's much better model than adding new concepts.
It's breaking change (i.e. behavior of host needs to change - in places where they were not passing .fluid, they would need to start passing .fluid), but it's so much simpler :)

Reply:
Considering this is going in 0.27, are we okay with it being a breaking change? Maybe not. Maybe in near future we remove default .fluid. What do you say @vladsud @leeviana

Reply:
It is a non required argument. Added undefined to the type.

Reply:
For now, I don't think we can have it be a breaking change as it would break all hosts when they pick up new bits. We can make our helper host logic code always default to .fluid though so eventually if all hosts are using our helper logic/copying our helper logic we can transition this API to required without breaking too many critical contracts.

Reply:
So right long term change is to drop driver adding extension, not adding new fields and complicating code when solution is one-liner.
if we want to add extension in back compat way in 0.27 - we can do it, but we should not bring it to main - main should have one-liner fix to drop .fluid extension and expect extension to be passed form host.

Reply:
So I will do these same changes in release/0.27 and here use the one liner change.

Reply:
Sounds good with me. Not ideal of course (we should figure out how not to need to bring changes like that to release branches), but working solution.